### PR TITLE
Revising taxi fare regression tutorial to .7

### PR DESCRIPTION
**Pull request by JRAlexander**
Revising taxi fare regression tutorial to upcoming ML .NET .7 API release

Code sample PR is https://github.com/dotnet/samples/pull/441

### CODE REVIEW

File: docs/machine-learning/tutorials/taxi-fare.md [hunk 1]
```diff
@@ -85,143 +85,143 @@ Remove the existing class definition and add the following code, which has two c
...
 
-> [!NOTE]
-> These steps must be added to the pipeline in the order specified above for successful execution.
+The last step in data preparation combines all of the feature columns into the **Features** column using the <xref:Microsoft.ML.Transforms.ColumnConcatenatingEstimator> transformation class. By default, a learning algorithm processes only features from the **Features** column. Add the following code:
 
-## Choose a learning algorithm
+[!code-csharp[ColumnConcatenatingEstimator](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#9 "Use the ColumnConcatenatingEstimator")]
 
-After adding the data to the pipeline and transforming it into the correct input format, you select a learning algorithm (**learner**). The learner trains the model. You chose a **regression** task for this problem, so you use a <xref:Microsoft.ML.Legacy.Trainers.FastTreeRegressor> learner, which is one of the regression learners provided by ML.NET.
+Notice that the `TripTime` column, which corresponds to the `trip_time_in_secs` column in the data set file, isn't included. We already determined that it isn't a useful prediction feature.
```
Comment (shmoradims) [inline docs/machine-learning/tutorials/taxi-fare.md]: >We already determined that it isn't a useful prediction feature. [](start = 125, length = 64)

I was able to use all the features with good results. Please try the following pipeline and compare the metrics. It should be much better. Of course, please change all the "new ..." to mlContext equivalent. There are examples in the cookbook.

var pipeline = new CopyColumnsEstimator(mlcontext, "FareAmount", "Label")
                                    .Append(new CategoricalEstimator(mlcontext, "VendorId"))
                                    .Append(new CategoricalEstimator(mlcontext, "RateCode"))
                                    .Append(new CategoricalEstimator(mlcontext, "PaymentType"))
                                    .Append(new Normalizer(mlcontext, "PassengerCount", Normalizer.NormalizerMode.MeanVariance))
                                    .Append(new Normalizer(mlcontext, "TripTime", Normalizer.NormalizerMode.MeanVariance))
                                    .Append(new Normalizer(mlcontext, "TripDistance", Normalizer.NormalizerMode.MeanVariance))
                                    .Append(new ConcatEstimator(mlcontext, "Features", "VendorId", "RateCode", "PassengerCount", "TripTime", "TripDistance", "PaymentType"));

File: docs/machine-learning/tutorials/taxi-fare.md [hunk 2]
```diff
@@ -85,143 +85,143 @@ Remove the existing class definition and add the following code, which has two c
...
 
 }
 ```
 
-The learning pipeline loads all of the data and algorithms necessary to train the model. Add the following code into the `Train` method:
-
-```csharp
-var pipeline = new LearningPipeline();
-```
+We are passing two parameters into the Train method; an MLContext for the context (`mlContext`), and a string for the dataset path (`dataPath`). We're going to reuse this method for loading datasets.
```
Comment (BillWagner) [inline docs/machine-learning/tutorials/taxi-fare.md]: Code fence `Train` and `MLContext`

File: docs/machine-learning/tutorials/taxi-fare.md [hunk 3]
```diff
@@ -85,143 +85,143 @@ Remove the existing class definition and add the following code, which has two c
...
 
-[!code-csharp[CreatePipeline](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#3 "Create and initialize the learning pipeline")]
+[!code-csharp[FastTreeRegressionTrainer](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#10 "Add the FastTreeRegressionTrainer")]
 
 ## Train the model
 
-The final step is to train the model. Until this point, nothing in the pipeline has been executed. The `pipeline.Train<TInput, TOutput>` method produces the model that takes in an instance of the `TInput` type and outputs an instance of the `TOutput` type. Add the following code into the `Train` method:
+The final step is to train the model. We train the model, <xref:Microsoft.ML.Runtime.Data.TransformerChain>, based on the dataset that has been loaded and transformed. Once the estimator has been defined, we train the model using the <xref:Microsoft.ML.Runtime.Data.EstimatorChain`1.Fit> while providing the already loaded training data. This returns a model to use for predictions. `pipeline.Fit()` trains the pipeline and returns a `Transformer` based on the `DataView` passed in. The experiment is not executed until this happens.
```
Comment (BillWagner) [inline docs/machine-learning/tutorials/taxi-fare.md]: You need the `%601` instead of the back tick.  Several locations with build warnings in this PR.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Overall, this is great @JRAlexander 

As in the previous tutorial, my only comments were for xref and related style concerns.

File: docs/machine-learning/tutorials/taxi-fare.md (hunk 1 above)
Comment (JRAlexander) [reply docs/machine-learning/tutorials/taxi-fare.md]: Will do. The cookbook was not updated when this was updated this weekend.

File: docs/machine-learning/tutorials/taxi-fare.md (hunk 3 above)
Comment (JRAlexander) [reply docs/machine-learning/tutorials/taxi-fare.md]: Yes, sorry, had not gotten to those yet.

Comment (JRAlexander) [reply docs/machine-learning/tutorials/taxi-fare.md]: The rest should resolve with API release

Comment (JRAlexander) [reply docs/machine-learning/tutorials/taxi-fare.md]: Clean build, @BillWagner!

File: docs/machine-learning/tutorials/taxi-fare.md [hunk 4]
```diff
@@ -85,143 +85,143 @@ Remove the existing class definition and add the following code, which has two c
...
 
-> [!NOTE]
-> These steps must be added to the pipeline in the order specified above for successful execution.
+The last step in data preparation combines all of the feature columns into the **Features** column using the `ColumnConcatenatingEstimator` transformation class. By default, a learning algorithm processes only features from the **Features** column. Add the following code:
 
-## Choose a learning algorithm
+[!code-csharp[ColumnConcatenatingEstimator](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#9 "Use the ColumnConcatenatingEstimator")]
 
-After adding the data to the pipeline and transforming it into the correct input format, you select a learning algorithm (**learner**). The learner trains the model. You chose a **regression** task for this problem, so you use a <xref:Microsoft.ML.Legacy.Trainers.FastTreeRegressor> learner, which is one of the regression learners provided by ML.NET.
+Notice that the `TripTime` column, which corresponds to the `trip_time_in_secs` column in the data set file, isn't included. We already determined that it isn't a useful prediction feature.
```
Comment (pkulikov) [inline docs/machine-learning/tutorials/taxi-fare.md]: Updated snippets include the "TripTime" column, so this paragraph doesn't match the code:
https://github.com/dotnet/samples/pull/441/files#diff-533141b9bd48d8d1591a35cdef83ab0fR80

Comment (JRAlexander) [reply docs/machine-learning/tutorials/taxi-fare.md]: @pkulikov - Good catch. Thanks!

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
The build is green 🚦   :shipit: when ready!

File: docs/machine-learning/tutorials/taxi-fare.md (hunk 4 above)
Comment (pkulikov) [reply docs/machine-learning/tutorials/taxi-fare.md]: @JRAlexander as this is already merged, another PR should be made to update the line 69. Now, it is:
>**trip_time_in_secs:** The amount of time the trip took. You want to predict the fare of the trip before the trip is completed. At that moment you don't know how long the trip would take. Thus, the trip time is not a feature and you'll exclude this column from the model.

**Pull request merged**